### DISCUSSION TITLE
Attaching change event listener on Toggle

### DISCUSSION THREAD

**Discussion by tgeorgiev** [vmware/clarity#6713] [Q&A] [answered]
From the available docs it is not clear how to attach change listeners to the `<cds-toggle>` component. 

Looking at the source there seems there is one private `checkedChange`, however it is emitted only when switching from checked state to unchecked, but not vice-versa
https://github.com/vmware/clarity/blob/efbf0f01432efb80a64fecb4bf021100687d49cf/packages/core/src/forms/control-inline/control-inline.element.ts#L109

For the time being I am attaching listeners to the `<input>` of the toggle, which does the job, is this the way to go?

**Comment by coryrylan** [comment, marked as answer]
Yes you would listen to the native input events. The benefit of projecting the input is it allows it to work like any other native html input. If you are using angular you can use angular forms or the event binding on the native change event.

https://stackblitz.com/edit/node-lckcvh?file=src%2Fapp%2Fapp.component.html

https://coryrylan.com/blog/building-forms-with-angular-and-clarity-design

**Reply by tgeorgiev** [reply]
Thank you @coryrylan for the prompt answer!
